packages/framework/src/Model/Product/Pricing/ProductManualInputPrice.php

```diff
@@ -96,16 +96,18 @@ public function recalculateInputPriceForNewVatPercent(
         BasePriceCalculation $basePriceCalculation,
         InputPriceCalculation $inputPriceCalculation
     ) {
-        $basePriceForPricingGroup = $basePriceCalculation->calculateBasePrice(
-            $this->getInputPrice(),
-            $inputPriceType,
-            $this->getProduct()->getVat()
-        );
-        $inputPriceForPricingGroup = $inputPriceCalculation->getInputPrice(
-            $inputPriceType,
-            $basePriceForPricingGroup->getPriceWithVat()->toValue(),
-            $newVatPercent
-        );
-        $this->setInputPrice(Money::fromValue($inputPriceForPricingGroup));
+        if ($this->inputPrice !== null) {
+            $basePriceForPricingGroup = $basePriceCalculation->calculateBasePrice(
+                $this->getInputPrice(),
```

Comment:
```suggestion
                $this->inputPrice,
```

This is better for static analysis - due to condition above, the `$this->inputPrice` cannot be `null` and can be safely passed to `calculateBasePrice`

Reply:
fixed in 7499b4024